diffkemp/output.py

```diff
@@ -1,10 +1,119 @@
+"""
+Contains logic for creating output of DiffKemp compare command.
+
+Specifically, the YAML file is created here which contains
+information about found differences and definitions of symbols
+which are located in the differences.
+"""
 from diffkemp.semdiff.caching import ComparisonGraph
 from diffkemp.utils import get_end_line, EndLineNotFound
 from diffkemp.semdiff.result import Result
+from enum import Enum
 import os
 import yaml
 
 
+class SymbolKind(str, Enum):
+    """Kind of symbol."""
+    FUNCTION = "function"
+    TYPE = "type"
+    MACRO = "macro"
+
+    def __str__(self):
+        return self.value
+
+
+class MacroDefinitions:
```

Comment:
Why is macro collection done using a class while function and types are handled directly by methods of `YamlOutput`. Are macro somewhat special that the collection must keep some state?

Reply:
> Why is macro collection done using a class while function and types are handled directly by methods of `YamlOutput`. Are macro somewhat special that the collection must keep some state?

The macro collection is done in a separate class mostly because the extraction of the definition is a little bit more complex so I created a class for it so the logic is encapsulated, also it was done in reaction to this comment on the original PR https://github.com/diffkemp/diffkemp/pull/314#discussion_r1517429792